Fixing the travis ci scala version. Also everything work on scala 2.12.7 now, scalaz concurrency is gone replaced with fs2 tasks.

GuyIncognito1986 opened the pull request.

Commit:
Fixing the client warnings/errors, upgrading libraries, fixing tests and making this build/work with scala 2.12

Commit:
Fixing the maybe stuff in requests, upgrading more libraries

Commit:
Swithching from scalaz concurrency to fs2 task, fixing all dependent code, tests still broken, but library now builds and we use the correct tasks

Commit:
fixing tests and test framework

Commit:
Fixed most of the tests, these never could have passed in the first place, whoever wrote these was a fucking moron

Commit:
only 16 tests now broken, mostly query stuff

Commit:
dbInfo serialization/deserialization now works properly, 14 tests failing now

Commit:
All tests outside query language now pass, WINZ

Commit:
All tests outside query language now pass, WINZ

Commit:
All tests outside query language now pass, WINZ

Commit:
Deleting temp views and temp view tests as couchdb2 no longer supports thems

Commit:
Deleting temp views and temp view tests as couchdb2 no longer supports thems

Commit:
final sbt changes for release

Commit:
adding stuff to publish to oss sonatype

Commit:
sonatype sbt changes for one click maven central release. WINZZZZZ

antonsemzy-svg committed:
Update README.md

antonsemzy-svg committed:
Update README.md

antonsemzy-svg committed:
Update README.md

antonsemzy-svg committed:
Update README.md

antonsemzy-svg committed:
Update README.md

Commit:
removing static libs and more temp view redundant code, updating readme

Commit:
Updating readme

Commit:
upgrading upickle to latest

Commit:
reverting upickle update because of breaking changes in the serializer

Commit:
CHanging travis scala version

beloglazov commented:
Thanks for all the changes! I'll start reviewing it soon. Some tests seem to be failing now?

GuyIncognito1986 commented:
Think they should all pass, Atleast they do on the local test environment,
I can imagine maybe we need to point them at whatever instance your Travis
pipeline points to? If you can give me the details I can update them. Also
that instance needs to be updated to couchdb 2.2, as I have a feeling its
going to be an old instance, some stuff like server info domain object have
changed etc.

On Thu, 15 Nov 2018, 12:32 Anton Beloglazov <notifications@github.com wrote:

> Thanks for all the changes! I'll start reviewing it soon. Some tests seem
> to be failing now?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/beloglazov/couchdb-scala/pull/80#issuecomment-439024829>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AYlhpapWyQ8NZu5uAM7yYHTxJRmURwR0ks5uvV7TgaJpZM4YblJ5>
> .
>

beloglazov commented:
Sorry for not getting back to this earlier. The PR in the current form is too large, and some of the changes are not acceptable, e.g., changing the package / license. I would accept smaller, more targeted PRs with clearer descriptions of what's being done (which would also make it easier and faster to review the changes).

giftig commented on README.md:
````diff
@@ -22,7 +16,7 @@ It's based on these awesome libraries:
 
 ```Scala
-libraryDependencies += "com.ibm" %% "couchdb-scala" % "0.7.2"
+libraryDependencies += "io.github.guyincognito1986" %% "couchdb-scala" % "1.0.3.1"
````
lol

giftig commented on README.md:
```diff
@@ -595,40 +602,12 @@ sbt "run-main com.ibm.couchdb.examples.Basic"
-
 ## Contributors
-
+  - Anton Semenov https://github.com/GuyIncognito1986
```
Cheeky.

GuyIncognito1986 commented:
Hey, I fixed the tests, switched in fs2 concurrency over scalaz and did a
whole bunch of stuff to make it work with scala 2.12, as well as release it
back into maven because those guys are not really supporting it anymore.
I think it's fair enough that I add myself as a contributor. (Also I had to
rename the package as I do not have maven rights over com.ibm in order to
release that package on maven central).

On Sun, 27 Jan 2019 at 15:16, Rob Moore <notifications@github.com> wrote:

> *@giftig* commented on this pull request.
> ------------------------------
>
> In README.md
> <https://github.com/beloglazov/couchdb-scala/pull/80#discussion_r251241277>
> :
>
> >  ## Contributors
> -
> +  - Anton Semenov https://github.com/GuyIncognito1986
>
> Cheeky.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/beloglazov/couchdb-scala/pull/80#pullrequestreview-196832013>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AYlhpS_hrM32pYIVi_KwCdtnBCkQPNA9ks5vHcK_gaJpZM4YblJ5>
> .
>

giftig commented:
You added yourself to the top of the contributor list, above the author, and messed with the licensing information.

As for the maven release, I assume @beloglazov intends to create a release including any patches people contribute. If what you're saying is that the author / IBM aren't maintaining it, why have you even created this PR to the original repo? If you intend to release it separately, no one's stopping you; do it from your fork (it's even namespaced **to** your fork)

GuyIncognito1986 commented:
On the off chance it would get reviewed and accepted so other people might
benefit from the changes. Modifying the licence info is a one liner, same
with moving my name to third or whatever. Plus I needed it for my startup,
so had to push to maven central pronto, as it's easier then dicking about
with local package repositories and sbt refs, and thought other people
might wana use it as there was interest in scala 2.12 support (and couchDb
2.2). And to be fair it took over a month to even get a reply from
Beloglazov.

On Sun, 27 Jan 2019 at 16:43, Rob Moore <notifications@github.com> wrote:

> You added yourself to the top of the contributor list, above the author,
> and messed with the licensing information.
>
> As for the maven release, I assume @beloglazov
> <https://github.com/beloglazov> intends to create a release including any
> patches people contribute. If what you're saying is that the author / IBM
> aren't maintaining it, why have you even created this PR to the original
> repo?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/beloglazov/couchdb-scala/pull/80#issuecomment-457933513>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AYlhpYaqn-r6HvO-NsQGZQ7FBNPYLjZHks5vHdcugaJpZM4YblJ5>
> .
>
